File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.java
class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        int n = nums.length;
        Map<Integer, Integer> mp = new HashMap<>();

        for(int i = 0; i<n; i++){
            int val = nums[i];
            if(mp.containsKey(val) && i - mp.get(val) <= k){
                return true;
            }
            mp.put(nums[i], i);
        }

        return false;
    }
}

File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.cpp
class Solution {
public:
    // Brute force approach TC{O(n^2)}
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        int n = nums.size();
        unordered_set<int> st;
        int i = 0, j = 0;
        while(j < n){
            if(abs(j-i) > k){
                st.erase(nums[i]);
                i++;
            }
            if(st.find(nums[j]) != st.end()){
                return true;
            }
            st.insert(nums[j]);
            j++;
        }
        return false;
    }
};